pkg/yamlgen/yamlgen_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package yamlgen
+
+import (
+	"testing"
+
+	"github.com/efficientgo/tools/core/pkg/testutil"
+	"golang.org/x/net/context"
+)
+
+func TestYAMLGen_GenGoCode(t *testing.T) {
+	t.Run("normal struct", func(t *testing.T) {
+		source := []byte("package main\n\ntype TestConfig struct {\n\tUrl   string `yaml:\"url\"`\n\tID    int    `yaml:\"id\"`\n\tToken string `yaml:\"token\"`\n}\n")
+		generatedCode, err := GenGoCode(source)
+		testutil.Ok(t, err)
+
+		expected := "package main\n\nimport (\n\t\"fmt\"\n\tyamlgen \"github.com/bwplotka/mdox/pkg/yamlgen\"\n\t\"os\"\n)\n\ntype TestConfig struct {\n\tUrl   string `yaml:\"url\"`\n\tID    int    `yaml:\"id\"`\n\tToken string `yaml:\"token\"`\n}\n\nfunc main() {\n\tconfigs := map[string]interface{}{}\n\tconfigs[\"TestConfig\"] = TestConfig{}\n\tfor k, config := range configs {\n\t\tfmt.Println(\"---\")\n\t\tfmt.Println(k)\n\t\tyamlgen.Generate(config, os.Stderr)\n\t}\n}\n"
+		testutil.Equals(t, expected, generatedCode)
+	})
+
+	t.Run("struct with unexported field", func(t *testing.T) {
+		source := []byte("package main\n\nimport \"regexp\"\n\ntype ValidatorConfig struct {\n\tType  string `yaml:\"type\"`\n\tRegex string `yaml:\"regex\"`\n\tToken string `yaml:\"token\"`\n\n\tr *regexp.Regexp\n}\n")
```

Comment:
I would use ` and add +  backtick + , something like that. Still easier than long string manually crafted.

Another idea - put all in files - that would do as well.